src/bls12_381/ec.rs

```diff
@@ -1653,6 +1733,17 @@ pub mod g2 {
         ::tests::curve::curve_tests::<G2>();
     }
 
+    #[test]
+    fn g2_test_hash() {
+        let mut rng = XorShiftRng::from_seed([0x5dbe6259, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
```

Comment:
A randomised test is good as far as it goes, but I'd also like to see tests that are selected to cover all code paths in the algorithm (and some cases that take the same path but for different reasons, such as when the input is zero).